### PR TITLE
Feat: support label filter for application list

### PR DESCRIPTION
**Pull request by basuotian**
### Description of your changes

Close: #240 

support label-based selector for application management, hiding label with prefix `ux.oam.dev` and `app.oam.dev`

image

in table mode, fold up labels in default

image

expand all labels

image

click label tag will invoke label filter

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary.
- [ ] Run `yarn lint` to ensure the frontend changes are ready for review.
- [ ] Run `make reviewable`to ensure the server changes are ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/server/interfaces/api/application.go [hunk 1]
```diff
@@ -687,11 +688,22 @@ func (c *application) listApplications(req *restful.Request, res *restful.Respon
...
+				labels[kv[0]] = kv[1]
+			}
+		}
+	}
 	apps, err := c.ApplicationService.ListApplications(req.Request.Context(), apis.ListApplicationOptions{
 		Projects:   projetNames,
 		Env:        req.QueryParameter("env"),
 		TargetName: req.QueryParameter("targetName"),
 		Query:      req.QueryParameter("query"),
+		Labels:     labels,
```
Comment (barnettZQG) [inline pkg/server/interfaces/api/application.go:706]: Please add an e2e test to cover this case.

File: packages/velaux-ui/src/locals/Zh/zh.json [hunk 2]
```diff
@@ -305,6 +305,7 @@
   "Search by Environment": "基于环境筛选",
   "Search by Target": "基于交付目标筛选",
   "Search by name and description etc": "基于名称、描述等筛选",
+  "Search by labels": "基于标签筛选",
```
Comment (wonderflow) [inline packages/velaux-ui/src/locals/Zh/zh.json:308]:
```suggestion
  "Search by Label Selector": "基于标签筛选",
```

**Review by wonderflow — COMMENTED** [review]
can I write a custom label selector?

**Comment by basuotian** [comment]
The label selector list comes from actual application labels, you can search with input content, nonexistent custom label selector would not get any applications

image

File: packages/velaux-ui/src/locals/Zh/zh.json [hunk 3]
```diff
@@ -619,5 +620,7 @@
   "contexts": "上下文参数",
   "Are you sure to rerun this Pipeline?": "确定要重新运行流水线吗？",
   "Show Values File": "查看 Value 文件",
-  "Showing the all projects that you have permissions": "你拥有权限的所有项目"
+  "Showing the all projects that you have permissions": "你拥有权限的所有项目",
```
[the remark this answers is not in the document]
Comment (basuotian) [reply packages/velaux-ui/src/locals/Zh/zh.json]: done

File: packages/velaux-ui/src/locals/Zh/zh.json [hunk 4]
```diff
@@ -305,6 +305,7 @@
   "Search by Environment": "基于环境筛选",
   "Search by Target": "基于交付目标筛选",
   "Search by name and description etc": "基于名称、描述等筛选",
```
[the remark this answers is not in the document]
Comment (basuotian) [reply packages/velaux-ui/src/locals/Zh/zh.json]: done

**Comment by wonderflow** [comment]
@basuotian The label selector could be very large scale, how do you get these labels? I'm very concerned about the memery usage.

**Comment by wonderflow** [comment]
But the current experience that we can auto-discovery the label selector is good

**Comment by basuotian** [comment]
> @basuotian The label selector could be very large scale, how do you get these labels? I'm very concerned about the memery usage.

Generated by traverse existing application list, the memory usage may be hard to avoid...i would make some test to ensure the memory usage

File: pkg/server/interfaces/api/application.go (hunk 1 above)
Comment (basuotian) [reply pkg/server/interfaces/api/application.go:706]: done, add case by using label selector to list applications

**Review by barnettZQG — APPROVED** [review]
LGTM

**Pull request merged**